store/store.go

```diff
@@ -190,7 +190,11 @@ func (s *DefaultStore) UpdateState(ctx context.Context, state types.State) error
 func (s *DefaultStore) GetState(ctx context.Context) (types.State, error) {
 	blob, err := s.db.Get(ctx, ds.NewKey(getStateKey()))
 	if err != nil {
-		return types.State{}, fmt.Errorf("failed to retrieve state: %w", err)
+		if err == ds.ErrNotFound {
```

Comment:
i don't think you need this parsing logic.`errors.Is(fmt.Errorf("failed to retrieve state: %w", ds.ErrNotFound), ds.ErrNotFound)` should still be `true`. can you double check?

Reply:
https://go.dev/play/p/iXeFMhNOb5M

@gupadhyaya is right, you don't need the parsing logic. See go playground example.